src/utils/coconut.js

```diff
@@ -0,0 +1,78 @@
+import 'brace/mode/python';
+
+export class CoconutHighlightRules extends window.ace.acequire(
+  'ace/mode/python_highlight_rules'
+).PythonHighlightRules {
+  constructor() {
+    super();
+
+    const keywords =
+      'and|as|assert|break|class|continue|def|del|elif|else|except|exec|' +
+      'finally|for|from|global|if|import|in|is|lambda|not|or|pass|print|' +
+      'raise|return|try|while|with|yield|async|await|data|' +
+      // Coconut-specific:
+      'match|case';
+    // TODO: Add rest
+
+    const builtinConstants =
+      'True|False|None|NotImplemented|Ellipsis|__debug__|';
+    // Coconut-specific:
+    // TODO: Add Coconut-specific builtin constants (if applicable?)
+
+    const builtinFunctions =
+      'abs|divmod|input|open|staticmethod|all|enumerate|int|ord|str|any|' +
+      'eval|isinstance|pow|sum|basestring|execfile|issubclass|print|super|' +
+      'binfile|iter|property|tuple|bool|filter|len|range|type|bytearray|' +
+      'float|list|raw_input|unichr|callable|format|locals|reduce|unicode|' +
+      'chr|frozenset|long|reload|vars|classmethod|getattr|map|repr|xrange|' +
+      'cmp|globals|max|reversed|zip|compile|hasattr|memoryview|round|' +
+      '__import__|complex|hash|min|set|apply|delattr|help|next|setattr|' +
+      'buffer|dict|hex|object|slice|coerce|dir|id|oct|sorted|intern|self|' +
+      // Coconut-specific
+      'reduce|takewhile|dropwhile|tee|count|makedata|consume|parallel_map|' +
+      'addpattern|recursive_iterator|concurrent_map|fmap|__fmap__|starmap|reiterable|' +
+      'scan|groupsof|py_chr|py_filter|py_hex|py_input|py_init|py_object|py_oct|' +
+      'py_open|py_print|py_range|py_xrange|py_str|py_map|py_zip|';
+
+    const keywordMapper = this.createKeywordMapper(
+      {
+        'invalid.deprecated': 'debugger',
+        'support.function': builtinFunctions,
```

Comment:
Add `"variable.language": "self|cls",` here to be aligned with https://github.com/ajaxorg/ace/blob/master/lib/ace/mode/python_highlight_rules.js

Reply:
Ok I added this. Should I put double instead of single quotes around "invalid.deprecated" : "debugger" and "constant.language": builtinConstants as well? Right now they are single quotes

Reply:
Single quotes but it doesn't really matter since I added a script that when you commit it gets formatted the right way.